code/datums/martial/boxing.dm

```diff
@@ -281,6 +296,66 @@
+		return FALSE
+
+	if(possible_boxer.mob_biotypes & honorable_mob_biotypes) //We're after animals, not people
+		return TRUE
+
+	return FALSE //rather than default assume TRUE, we default assume FALSE. After all, there could be mobs that are none of our biotypes and also not humanoid. By default, they would be valid for being boxed if TRUE.
+
+// Our hunter boxer applies a rebuke and double damage against the target of their crit. If the target is humanoid, we just perform our regular crit effect instead.
+
+/datum/martial_art/boxing/hunter/crit_effect(mob/living/attacker, mob/living/defender, armor_block = 0, damage_type = STAMINA, damage = 0)
+	if(defender.mob_biotypes & MOB_HUMANOID && !istype(defender, /mob/living/simple_animal/hostile/megafauna))
+		return ..() //Applies the regular crit effect if it is a normal human, and not a megafauna
+
+	var/first_word_strike = pick("Extinction", "Brutalization", "Explosion", "Adventure", "Thunder", "Lightning", "Sonic", "Atomizing", "Whirlwind", "Tornado", "Shark", "Falcon")
+	var/second_word_strike = pick(" Punch", " Pawnch", "-punch", " Jab", " Hook", " Fist", " Uppercut", " Straight")
+	defender.visible_message(
+		span_danger("[attacker] knocks the absolute bajeezus out of [defender] utilizing the terrifying [first_word_strike][second_word_strike]!!!"),
+		span_userdanger("You have the absolute bajeezus knocked out of you by [attacker]!!!"),
+		span_hear("You hear a sickening sound of flesh hitting flesh!"),
+		COMBAT_MESSAGE_RANGE,
+		attacker,
+	)
+	to_chat(attacker, span_danger("You knock the absolute bajeezus out of [defender] out with the terrifying [first_word_strike][second_word_strike]!!!"))
+	if(ishuman(attacker))
+		var/mob/living/carbon/human/human_attacker = attacker
+		human_attacker.force_say()
+		human_attacker.say("[first_word_strike][second_word_strike]!!!", forced = "hunter boxing enthusiastic battlecry")
+	defender.apply_status_effect(/datum/status_effect/rebuked)
+	defender.apply_damage(damage * 2, default_damage_type, BODY_ZONE_CHEST, armor_block) //deals double our damage AGAIN
+	attacker.reagents.add_reagent(/datum/reagent/medicine/omnizine/godblood, 3) //Get a little healing in return for a successful crit
+	log_combat(attacker, defender, "hunter crit punched (boxing)")
+
+// Our hunter boxer speeds up their attacks when completing a combo against a valid target, and does a sizable amount of extra damage.
+
+/datum/martial_art/boxing/hunter/perform_extra_effect(mob/living/attacker, mob/living/defender)
+	if(defender.mob_biotypes & MOB_HUMANOID && !istype(defender, /mob/living/simple_animal/hostile/megafauna))
+		return // Does not apply to humans (who aren't megafauna)
+
+	defender.changeNext_move(CLICK_CD_RAPID)
+	defender.apply_damage(rand(15,20), default_damage_type, BODY_ZONE_CHEST)
```

Comment:
These almost certainly shouldn't _both_ be `defender`, I suspect the first one is wrong and you are actually lowering bubblegum's click CD when you punch him (or increasing it, if he wasn't already on click CD?)

Reply:
Oh whoops thats a my bad